timely/src/dataflow/operators/core/reclock.rs

```diff
@@ -57,7 +57,9 @@ impl<S: Scope, C: Container> Reclock<S> for StreamCore<S, C> {
 
             // stash each data input with its timestamp.
             input1.for_each(|cap, data| {
-                stash.push((cap.time().clone(), std::mem::take(data)));
+                for data in data {
+                    stash.push((cap.time().clone(), std::mem::take(data)));
+                }
```

Comment:
This is a moment where the un-grouped iterator could have helped the migration, as we seem to want a capability per container.